corehq/apps/userreports/reports/filters/factory.py

```diff
@@ -86,6 +86,22 @@ def _build_dynamic_choice_list_filter(spec, report):
     )
 
 
+def _build_multi_field_dynamic_choice_list_filter(spec, report):
+    wrapped = MultiFieldDynamicChoiceFilterSpec.wrap(spec)
+    choice_provider_spec = wrapped.get_choice_provider_spec()
+    choice_provider = MultiFieldChoiceProviderFactory.from_spec(choice_provider_spec)(report, wrapped.slug)
+    choice_provider.configure(choice_provider_spec)
+    return DynamicChoiceListFilter(
+        name=wrapped.slug,
+        datatype=wrapped.datatype,
+        field=wrapped.fields,
```

Comment:
I know that this looks weird but this field is used here https://github.com/dimagi/commcare-hq/pull/16285/files#diff-fe5cacb99e1e5c5f67991de667b352cdR195